src/featuremap/ugbio_featuremap/somatic_featuremap_utils.py

```diff
@@ -0,0 +1,155 @@
+    with open(sorted_tsv, "w") as out_file:
+        subprocess.check_call(cmd, stdout=out_file)
+
+    gz_tsv = sorted_tsv + ".gz"
+    cmd = ["bgzip", "-c", sorted_tsv]
+    with open(gz_tsv, "wb") as out_file:
+        subprocess.check_call(cmd, stdout=out_file)
+
+    cmd = ["tabix", "-s", "1", "-b", "2", "-e", "2", gz_tsv]
+    subprocess.check_call(cmd)
+
+    hdr_txt = [
+        '##INFO=<ID=TR_START,Number=1,Type=String,Description="Closest tandem Repeat Start">',
+        '##INFO=<ID=TR_END,Number=1,Type=String,Description="Closest Tandem Repeat End">',
+        '##INFO=<ID=TR_SEQ,Number=1,Type=String,Description="Closest Tandem Repeat Sequence">',
+        '##INFO=<ID=TR_DISTANCE,Number=1,Type=String,Description="Closest Tandem Repeat Distance">',
+        '##INFO=<ID=TR_LENGTH,Number=1,Type=String,Description="Closest Tandem Repeat total length">',
+        '##INFO=<ID=TR_SEQ_UNIT_LENGTH,Number=1,Type=String,Description="Closest Tandem Repeat unit length">',
+    ]
+    hdr_file = pjoin(tmpdir, "tr_hdr.txt")
+    with open(hdr_file, "w") as f:
+        f.writelines(line + "\n" for line in hdr_txt)
+
+    return gz_tsv, hdr_file
+
+
+def integrate_tandem_repeat_features(merged_vcf, ref_tr_file, out_dir):
+    """
+    Integrate tandem repeat features into a VCF file.
+
+    This function annotates variants in a VCF file with information about nearby tandem repeats,
+    including their start/end positions, sequences, distances, and lengths.
+
+    Parameters
+    ----------
+    merged_vcf : str
+        Path to the input VCF file (gzipped) containing variants to be annotated.
+    ref_tr_file : str
+        Path to the reference tandem repeat BED file containing tandem repeat regions.
+    out_dir : str
+        Output directory where the annotated VCF file and temporary files will be written.
+
+    Returns
+    -------
+    str
+        Path to the output VCF file with tandem repeat annotations. The output file will be
+        gzipped and indexed, with '.tr_info.vcf.gz' suffix replacing the original '.vcf.gz'.
+
+    Notes
+    -----
+    The function adds the following INFO fields to the VCF:
+    - TR_START: Start position of the closest tandem repeat
+    - TR_END: End position of the closest tandem repeat
+    - TR_SEQ: Sequence of the tandem repeat unit
+    - TR_DISTANCE: Distance from variant to the closest tandem repeat
+    - TR_LENGTH: Total length of the tandem repeat region
+    - TR_SEQ_UNIT_LENGTH: Length of the tandem repeat unit
+    """
+    with tempfile.TemporaryDirectory(dir=out_dir) as tmpdir:
+        # Create variant BED file
+        bed1 = pjoin(tmpdir, "merged_vcf.tmp.bed")
+        _create_variant_bed(merged_vcf, bed1)
+
+        # Sort reference tandem repeat file
+        ref_tr_file_sorted = pjoin(tmpdir, "ref_tr_file.sorted.bed")
+        cmd = ["bedtools", "sort", "-i", ref_tr_file]
+        with open(ref_tr_file_sorted, "w") as sorted_file:
+            subprocess.check_call(cmd, stdout=sorted_file)
```

Comment:
Why is this part needed? If I understand correctly you start with a properly sorted file (same order as the vcf) and reorder it lexicographically to 
chr1
chr10
chr11
...

Reply:
removed this part. presorted and precalculated the tandem-repeat reference file.